apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaProducerInterceptor.java

```diff
@@ -59,8 +59,12 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             next = next.next();
             record.headers().add(next.getHeadKey(), next.getHeadValue().getBytes());
         }
-        EnhancedInstance callbackInstance = (EnhancedInstance) allArguments[1];
-        if (null != callbackInstance) {
+        //fix bug: https://github.com/apache/skywalking/issues/4825
```

Comment:
We don't add this link, once we did that, the links are everywhere.

Reply:
fixd.